components/authz-service/engine/conformance/conformance_v2_test.go

```diff
@@ -262,10 +263,122 @@ func TestV2ProjectsAuthorized(t *testing.T) {
 					},
 				}
 				setPoliciesV2(t, e, pol)
-				actual, err := e.V2ProjectsAuthorized(args())
+				actual, err := e.V2ProjectsAuthorized(args([]string{proj1, proj2}))
 				require.NoError(t, err)
 				assert.ElementsMatch(t, []string{proj2}, actual)
 			})
+
+			t.Run("policy that denies all projects and allows one project returns no projects", func(t *testing.T) {
+				pol := map[string]interface{}{
+					"members": engine.Subject(sub),
+					"statements": map[string]interface{}{
+						"statement-id-0": map[string]interface{}{
+							"actions":   []string{act},
+							"resources": []string{res},
+							"effect":    "deny",
+							"projects":  []string{constants.AllProjectsID},
+						},
+						"statement-id-1": map[string]interface{}{
+							"actions":   []string{act},
+							"resources": []string{res},
+							"effect":    "allow",
+							"projects":  []string{proj1},
+						},
+					},
+				}
+				setPoliciesV2(t, e, pol)
+				actual, err := e.V2ProjectsAuthorized(args([]string{proj1, proj2}))
+				require.NoError(t, err)
+				assert.ElementsMatch(t, []string{}, actual)
+			})
+
+			t.Run("policy that denies all projects returns no projects when request contains no projects", func(t *testing.T) {
```

Comment:
Similar to above comment, we'd be better served by changing the name:
s/when request contains no projects/when all projects requested/